File: scripts/events-between.py
#!/usr/bin/python

import re
import sys

if len(sys.argv) != 4:
    sys.exit("usage: ./events-between.py <file> <start> <stop>")

f = open(sys.argv[1])
for line in f.readlines():
    line = line.strip()
    if line[0] == 'h' or line[0] == 'p':
        print line
    elif line[0] == 'e' or line[0] == 'b':
        m = re.match(r'(?:[^;]+;){3}([^;]+);', line)
        if int(sys.argv[2]) <= int(m.group(1)) and int(sys.argv[3]) >= int(m.group(1)):
            print line
    else:
        sys.exit("unable to parse line %s" % line)
f.close()

File: scripts/events-between.rb
#!/usr/bin/env ruby

if ARGV.length != 3
    puts "usage: ./events-between.rb <file> <start> <stop>"
end

lines = IO.readlines ARGV.shift

min = ARGV.shift.to_i
max = ARGV.shift.to_i

lines.each do |line|
    # take care of line endings
    line.chomp!.chomp!
    
    a = line.match /([^;]+);(?:([^;]+);([^;]+);([^;]+);)?/
    
    case a[1]
    when 'h'
        puts line
    when 'p'
        puts line
    when 'e'
        if min <= a[4].to_i and max >= a[4].to_i
            puts line
        end
    when 'b'
        if min <= a[4].to_i and max >= a[4].to_i
            puts line
        end        
    else
        abort "unable to parse line: #{line}"    
    end
end
